j2cl-tasks/src/main/java/com/vertispan/j2cl/build/provided/J2clTask.java

```diff
@@ -84,6 +93,26 @@ public Task resolve(Project project, Config config) {
             if (!j2cl.transpile(javaSources, nativeSources)) {
                 throw new IllegalStateException("Error while running J2CL");
             }
+
+            if(sourcemapsEnabled) {
+                if(sourceMapsFolder.toFile().exists()) {
+                    FileUtils.deleteDirectory(sourceMapsFolder.toFile());
+                }
+                sourceMapsFolder.toFile().mkdirs();
+                FileUtils.copyDirectory(context.outputPath().toFile(), sourceMapsFolder.toFile());
+            }
         };
     }
+
+    private Path prepareSourcesFolder(Config config, Project project) {
```

Comment:
Won't doing this here prevent projects that don't get j2cl run on them (bootstrap, jre, any JS-only dependency) from getting sourcemaps? It seems like this might belong in ClosureBundleTask instead (copy from "here's where my input declared its sourcemaps, skip ahead and copy straight to the output dir).

That would also remove the need to have a "is compilationLevel==BUNDLE_JAR", which technically this should have as written - if you have this property on when doing a prod build, you're going to write a bunch of extra stuff to prod that you probably don't want...

Reply:
> Won't doing this here prevent projects that don't get j2cl run on them (bootstrap, jre, any JS-only dependency) from getting sourcemaps? It seems like this might belong in ClosureBundleTask instead (copy from "here's where my input declared its sourcemaps, skip ahead and copy straight to the output dir).
> 
> That would also remove the need to have a "is compilationLevel==BUNDLE_JAR", which technically this should have as written - if you have this property on when doing a prod build, you're going to write a bunch of extra stuff to prod that you probably don't want...

Definitely, what's good is that the projects you mentioned are those that won't change during the j2cl:watch process, which means their source maps need to be copied only once. ClosureBundleTask is definitely the best place for this.

I was thinking about how to copy the source maps of reactor modules in ClosureBundleTask, rather than in J2clTask, but I couldn't figure out how to let ClosureBundleTask know which projects have changed and which have not. Maybe you have some ideas?